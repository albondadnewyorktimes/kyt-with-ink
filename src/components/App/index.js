import React from 'react';
import { Switch, Route, Link } from 'react-router-dom';
import dynamic from 'kyt-runtime/dynamic';
import { Button } from '@nyt-cms/ink';
import * as styles from './styled';
import '@nyt-cms/ink/lib/tokens/typography/fonts';

export const Home = dynamic(() => import(/* webpackChunkName: "home" */ '../Home'));
export const Tools = dynamic(() => import(/* webpackChunkName: "tools" */ '../Tools'));

function App() {
  return (
    <div>
      <i className={styles.logoClass} />
      <ul className={styles.navClass}>
        <li className={styles.navItemClass}>
          <Link className={styles.linkClass} to="/">
            Home
          </Link>
        </li>
        <li className={styles.navItemClass}>
          <Link className={styles.linkClass} to="/tools">
            Tools
          </Link>
        </li>
      </ul>
      <div className={styles.contentClass}>
        <Switch>
          <Route path="/" exact component={Home} />
          <Route path="/tools" component={Tools} />
        </Switch>
        <br />
        <Button>Hello world!</Button>
      </div>
    </div>
  );
}

export default App;
