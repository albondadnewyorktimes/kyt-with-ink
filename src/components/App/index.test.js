import React from 'react';
import { render } from '@testing-library/react';
import { StaticRouter } from 'react-router-dom';
import { DynamicImports } from 'kyt-runtime/server';
import App, { Home, Tools } from '.';

describe('App', () => {
  it('Test home route', async () => {
    await Home.preload();
    const modules = [];
    const { container } = render(
      <DynamicImports report={module => modules.push(module)}>
        <StaticRouter location="/" context={{}}>
          <App />
        </StaticRouter>
      </DynamicImports>
    );

    expect(container.firstChild).toMatchSnapshot();
    expect(modules).toMatchSnapshot();
  });

  it('Test tools route', async () => {
    await Tools.preload();
    const modules = [];
    const { container } = render(
      <DynamicImports report={module => modules.push(module)}>
        <StaticRouter location="/tools" context={{}}>
          <App />
        </StaticRouter>
      </DynamicImports>
    );

    expect(container.firstChild).toMatchSnapshot();
    expect(modules).toMatchSnapshot();
  });
});
