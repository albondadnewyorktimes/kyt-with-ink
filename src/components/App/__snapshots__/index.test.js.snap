// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test home route 1`] = `
.lights-5 {
  width: 500px;
  margin: 0 auto;
  padding: 15px;
  text-align: center;
}

.lights-8 {
  width: 500px;
  margin: 15px auto;
}

.lights-2 {
  display: inline-block;
}

.lights-1 {
  padding: 15px;
  color: #00a68f;
  font-size: 18px;
}

.lights-0 {
  display: block;
  width: 252px;
  height: 252px;
  margin: 15px auto;
  background: url(file-stub);
}

.lights-6 + .lights-6 {
  padding-top: 15px;
}

<div>
  <i
    class="lights-0"
  />
  <ul
    class="lights-5"
  >
    <li
      class="lights-2"
    >
      <a
        class="lights-1"
        href="/"
      >
        Home
      </a>
    </li>
    <li
      class="lights-2"
    >
      <a
        class="lights-1"
        href="/tools"
      >
        Tools
      </a>
    </li>
  </ul>
  <div
    class="lights-8"
  >
    <section>
      <p
        class="lights-6"
      >
        Welcome to the 
        <strong>
          Universal React Starter-kyt
        </strong>
        . This starter kyt should serve as the base for an advanced, server-rendered React app.
      </p>
      <p
        class="lights-6"
      >
        Check out the Tools section for an outline of the libraries that are used in this Starter-kyt.
      </p>
    </section>
  </div>
</div>
`;

exports[`App Test home route 2`] = `
Array [
  "../Home",
]
`;

exports[`App Test tools route 1`] = `
.lights-5 {
  width: 500px;
  margin: 0 auto;
  padding: 15px;
  text-align: center;
}

.lights-11 {
  width: 500px;
  margin: 15px auto;
}

.lights-2 {
  display: inline-block;
}

.lights-1 {
  padding: 15px;
  color: #00a68f;
  font-size: 18px;
}

.lights-0 {
  display: block;
  width: 252px;
  height: 252px;
  margin: 15px auto;
  background: url(file-stub);
}

.lights-6 + .lights-6 {
  padding-top: 5px;
}

<div>
  <i
    class="lights-0"
  />
  <ul
    class="lights-5"
  >
    <li
      class="lights-2"
    >
      <a
        class="lights-1"
        href="/"
      >
        Home
      </a>
    </li>
    <li
      class="lights-2"
    >
      <a
        class="lights-1"
        href="/tools"
      >
        Tools
      </a>
    </li>
  </ul>
  <div
    class="lights-11"
  >
    <ul>
      <li
        class="lights-6"
      >
        <a
          href="https://expressjs.com/"
        >
          Express
        </a>
         - server-side rendering
      </li>
      <li
        class="lights-6"
      >
        <a
          href="https://reactjs.org/"
        >
          React
        </a>
         - component library
      </li>
      <li
        class="lights-6"
      >
        <a
          href="https://reacttraining.com/react-router/"
        >
          React Router
        </a>
         - server and browser routing
      </li>
      <li
        class="lights-6"
      >
        <a
          href="https://github.com/nytimes/pretty-lights"
        >
          Pretty Lights
        </a>
         - CSS-in-JS with a reliable API
      </li>
      <li
        class="lights-6"
      >
        <a
          href="https://testing-library.com/docs/react-testing-library/intro/"
        >
          React Testing Library
        </a>
         
        - React component testing
      </li>
    </ul>
  </div>
</div>
`;

exports[`App Test tools route 2`] = `
Array [
  "../Tools",
]
`;
